src/blenderbim/blenderbim/bim/__init__.py

```diff
@@ -67,8 +67,38 @@
     "debug": None,
 }
 
+dynamic_operators = [] #temporary list to hold dynamically generated operators
+
 for name in modules.keys():
     modules[name] = importlib.import_module(f"blenderbim.bim.module.{name}")
+    nameUpperCase = name[0].upper()+name[1:]
+
+    #generate operators that toggle module state
+    idname = f"bim.toggle_module_{name}"
+    
+    def func(self, context):
+        try:
+            for cls in self.classes:
+                bpy.utils.unregister_class(cls)
+                #self.is_active = False
+        except:
+            #class is already registered
+            for cls in self.classes:
+                bpy.utils.register_class(cls)
+                #self.is_active = True
+        return {'FINISHED'}
+
+    opclass = type(
+        f"BIM_OT_Toggle_Module_{nameUpperCase}",
+        (bpy.types.Operator, ),
+        {
+        "bl_idname": idname, 
+        "bl_label": nameUpperCase, 
+        "execute": func,
+        "classes": modules[name].classes
```

Comment:
I think it's clever to parse all classes in specific modules, but I think we're going to run into problems when some modules will try to comunicate with other modules. Maybe we can restrict classes to only those of type `bpy.types.Panel` ? That way only UI elements are unregistered and operators can still be called by other modules ?

Reply:
That`s a great idea, I was worrying about that too.  I´ll change it in the next commit.